float32 target_rf
float32 target_rb
float32 target_lf
float32 target_lb
int32 rf_count
int32 rb_count
int32 lf_count
int32 lb_count